tools/build/token-usage-detector.js

```diff
@@ -0,0 +1,153 @@
+const path = require('path');
```

Comment:
In a super ideal world this would be typescript and have tests. I'd be ok with just tests for now.

Reply:
Take a look again. Full TS and tests added.